server/handler/component.go

```diff
@@ -80,20 +97,58 @@ func (h *componentSigner) sign(r *http.Request, m *model.ComponentManifest) (sr
 		}
 
 		var indexVersion uint
-		var keys map[string]*v1manifest.KeyInfo
+		var owner *v1manifest.Owner
 		if err := md.UpdateIndexManifest(initTime, func(om *model.IndexManifest) *model.IndexManifest {
-			om.Signed.Components[name] = v1manifest.ComponentItem{
-				Owner: "pingcap", // TODO: read this from request
-				URL:   fmt.Sprintf("/%s.json", name),
+			// We only update index.json when it's a new component
+			// or the yanked, standalone, hidden fileds changed
+			var (
+				compItem  v1manifest.ComponentItem
+				compExist bool
+			)
+
+			if compItem, compExist = om.Signed.Components[name]; compExist {
+				// Find the owner of target component
+				o := om.Signed.Owners[compItem.Owner]
+				owner = &o
+				if len(options) == 0 {
+					// No changes on index.json
+					indexVersion = om.Signed.Version
+					return nil
+				}
+				if opt, ok := options["yanked"]; ok {
+					compItem.Yanked = opt
+				}
+				if opt, ok := options["hidden"]; ok {
+					compItem.Hidden = opt
+				}
+				if opt, ok := options["standalone"]; ok {
+					compItem.Standalone = opt
+				}
+			} else {
+				var ownerID string
+				// The component is a new component, so the owner is whoever first create it.
+				for _, sk := range m.Signatures {
+					if ownerID, owner = om.KeyOwner(sk.KeyID); owner != nil {
+						break
+					}
+				}
+				compItem = v1manifest.ComponentItem{
+					Owner:      ownerID,
+					URL:        fmt.Sprintf("/%s.json", name),
+					Yanked:     options["yanked"],
+					Standalone: options["standalone"],
+					Hidden:     options["hidden"],
+				}
 			}
-			keys = om.Signed.Owners["pingcap"].Keys
+
+			om.Signed.Components[name] = compItem
 			indexVersion = om.Signed.Version + 1
```

Comment:
The `indexVersion = om.Signed.Version + 1` only be applied if `compExist = false`. It's better to move the L145 to  above L141.

Reply:
We must update component.json when user try to modify that component. Otherwise, people can tamper others' components because they don't need to sign to modify the attributes(hidden, standalone, yanked)